horovod/spark/lightning/remote.py

```diff
@@ -106,22 +103,25 @@ def train(serialized_model):
             if train_logger is None:
                 train_logger = TensorBoardLogger(logs_path)
 
-            # TODO: find out a way to use ckpt_path created from remote store, but all other parameters ingest from estimator config
-            # ckpt_path = os.path.join(run_output_dir, remote_store.checkpoint_filename)
-            # os.makedirs(ckpt_path, exist_ok=True)
-            # model_checkpoint_callback = ModelCheckpoint(dirpath=ckpt_path)
-            # callbacks.append(model_checkpoint_callback)
-
-            is_model_checkpoint_callback_exist = False
-            for cb in callbacks:
-                if isinstance(cb, ModelCheckpoint):
-                    is_model_checkpoint_callback_exist = True
-                    break
+            # Lightning requires to add cehckpoint callbacks for all ranks.
+            # Otherwise we are seeing hanging in training.
+            _checkpoint_callback = checkpoint_callback
+            if _checkpoint_callback:
+                _checkpoint_callback.dir_path = ckpt_dir
+                _checkpoint_callback.filename = ckpt_filename
+            else:
+                # By default 'monitor'=None which saves a checkpoint only for the last epoch.
```

Comment:
in this case, `_checkpoint_callback.best_model_path` is the last model check point right?

Reply:
yes

Reply:
explanation in PL: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/callbacks/model_checkpoint.py#L53
```
After training finishes, use :attr:`best_model_path` to retrieve the path to the
    best checkpoint file
```